web/src/features/fba/components/ZoneSummaryPanel.tsx

```diff
@@ -1,30 +1,183 @@
 import React from 'react'
-import { TextField } from '@mui/material'
+import { Grid, Paper, Typography } from '@mui/material'
 import makeStyles from '@mui/styles/makeStyles'
-import { isNull } from 'lodash'
+import { isUndefined } from 'lodash'
+import { FireZone, FireZoneThresholdFuelTypeArea } from 'api/fbaAPI'
+import { PieChart, Pie, ResponsiveContainer, Cell } from 'recharts'
 
 const useStyles = makeStyles({
   wrapper: {
-    minWidth: 300
+    minWidth: 400
+  },
+  zoneName: {
+    fontSize: '2rem',
+    textAlign: 'center',
+    variant: 'h2'
+  },
+  centreName: {
+    fontSize: '1rem',
+    textAlign: 'right',
+    variant: 'h6'
+  },
+  fuelTypesPaper: {
+    padding: '20px 10px'
+  },
+  fuelTypesHeader: {
+    fontSize: '1.3rem',
+    textAlign: 'center',
+    variant: 'h3'
+  },
+  pieChartHeader: {
+    fontSize: '1rem',
+    variant: 'h4'
   }
 })
 
 interface Props {
   className?: string
-  selectedZoneID: number | null
+  selectedFireZone: FireZone | undefined
+  fuelTypeInfo: Record<number, FireZoneThresholdFuelTypeArea[]>
 }
 
+interface FuelTypeDataForPieChart {
+  area: number
+  fuel_type_code: string
+}
+
+const RADIAN = Math.PI / 180
+const COLOURS = [
+  '#2191FB',
+  '#FCB1A6',
+  '#B33951',
+  '#CCF5AC',
+  '#8CDEDC',
+  '#9DACFF',
+  '#4F7CAC',
+  '#FFA62B',
+  '#C09BD8',
+  '#EBC3DB',
+  '#D19C1D',
+  '#FFC0BE',
+  '#ED7D3A'
+]
+
 const ZoneSummaryPanel = (props: Props) => {
   const classes = useStyles()
 
-  if (isNull(props.selectedZoneID)) {
+  const renderCustomizedLabel = ({
+    cx,
+    cy,
+    midAngle,
+    innerRadius,
+    outerRadius,
+    percent,
+    fuel_type_code,
+    area,
+    index
+  }: {
+    cx: number
+    cy: number
+    midAngle: number
+    innerRadius: number
+    outerRadius: number
+    percent: number
+    fuel_type_code: string
+    area: number
+    index: number
+  }) => {
+    const radius = innerRadius + (outerRadius - innerRadius) * 0.5
+    const x = cx + radius * Math.cos(-midAngle * RADIAN)
+    const y = cy + radius * Math.sin(-midAngle * RADIAN)
+
+    if (percent * 100 < 2) {
+      return
+    }
+    if (percent * 100 < 5) {
+      return (
+        <text x={x} y={y} fontSize={'10pt'} fill="black">
+          {`${fuel_type_code} (${(percent * 100).toFixed(0)}%)`}
+        </text>
+      )
+    }
+
+    return (
+      <text x={x} y={y} fontSize={'10pt'} fill="black" textAnchor={x > cx ? 'start' : 'end'}>
+        {`${fuel_type_code}: ${area.toLocaleString(undefined, { maximumFractionDigits: 0 })} ha (${(
+          percent * 100
+        ).toFixed(0)}%)`}
+      </text>
+    )
+  }
+
+  if (isUndefined(props.selectedFireZone) || isUndefined(props.fuelTypeInfo[props.selectedFireZone.mof_fire_zone_id])) {
     return <div></div>
   } else {
+    const advisories: FuelTypeDataForPieChart[] = []
+    const warnings: FuelTypeDataForPieChart[] = []
+    props.fuelTypeInfo[props.selectedFireZone?.mof_fire_zone_id].forEach(thing => {
```

Comment:
Can we rename `thing` to something more descriptive?

Reply:
HAH. whoops. I intended to go back and rename that but then forgot!